README.md

```diff
@@ -1,5 +1,79 @@
+<p align="center">
+<img src="https://github.com/googlesamples/android-architecture/wiki/images/aab-logov2.png" alt="Android Architecture Blueprints"/>
+</p>
 
+Android Architecture Blueprints is a project to showcase different architectural approaches to developing Android apps. In its different branches you'll find the same app (a TODO app) implemented with small differences.
+
+In this master branch you'll find:
+*   Kotlin **[Coroutines](https://kotlinlang.org/docs/reference/coroutines-overview.html)** for background operations.
+*   A single-activity architecture, using the **[Navigation component](https://developer.android.com/guide/navigation/navigation-getting-started)** to manage fragment operations.
+*   A presentation layer that contains a fragment (View) and a **ViewModel** per screen (or feature).
+*   Reactive UIs using **LiveData** observables and **Data Binding**.
+*   A **data layer** with a repository and two data sources (local using Room and remote) that are queried with one-shot operations (no listeners or data streams).
+*   Two **product flavors**, `mock` and `prod`, [to ease development and testing](https://android-developers.googleblog.com/2015/12/leveraging-product-flavors-in-android.html) (except in the Dagger branch)
+*   A collection of unit, integration and e2e **tests**, including "shared" tests that can be run on emulator/device or Robolectric.
+
+## Variations
+
+This project hosts each sample app in separate repository branches. For more information, see the `README.md` file in each branch.
+
+### Stable samples - Kotlin
+| Sample | Description |
+| ------------- | ------------- |
+| [master](/) | The base for the rest of the branches. Uses Kotlin, Architecture Components, coroutines, Data Binding, etc. |
+| [dagger](tree/dagger/) | A simple Dagger setup that uses `dagger-android` and removes the two flavors. |
+| [usecases](tree/usecases/) | Adds a new domain layer that uses UseCases for business logic. |
+
+### Samples in development - Kotlin
+
+| Sample | Description |
+| ------------- | ------------- |
+| [reactive](tree/dagger/) | Modifies the data layer so UIs can react to changes automatically using Room as source of truth. |
+
+### Old samples - Kotlin and Java
+
+Blueprints v1 had a collection of samples that are not maintained anymore, but can still be useful. See [all project branches](https://github.com/googlesamples/android-architecture/branches).
+
+## Why a to-do app?
+
+<img align="right" src="https://github.com/googlesamples/android-architecture/wiki/images/todoapp.gif" alt="A demo illustraating the UI of the app" width="288" height="512" style="display: inline; float: right"/>
+
+The app in this project aims to be simple enough that you can understand it quickly, but complex enough to showcase difficult design decisions and testing scenarios. For more information, see the [app's specification](https://github.com/googlesamples/android-architecture/wiki/To-do-app-specification).
```

Comment:
```suggestion
The app in this project aims to be small enough to navigate quickly while being complex enough to showcase difficult design decisions and testing scenarios. For more information, see the [app's specification](https://github.com/googlesamples/android-architecture/wiki/To-do-app-specification).
```

Reply:
I'd prefer to keep this paragraph as it was written by a TW.

Reply:
SGTM